### ISSUE TITLE
Spring Controller annotated with @JsonView and hibernate4

### ISSUE THREAD

**Issue by bademux** [FasterXML/jackson-datatype-hibernate#73]
Hello,
It isn't a bug, bug nasty behavior.
I configure my MappingJackson2HttpMessageConverter for using ObjectMapper and Hibernate4Module with SessionFactory
https://gist.github.com/bademux/8e084d56c7aae2283d66

Spring along @JsonView-aware Controller has been used
see for more info http://docs.spring.io/spring-framework/docs/current/spring-framework-reference/html/new-in-4.1.html#_web_improvements

I supposed that all Lazy fields will be initiated and serialized, but got "could not initialize proxy - no Session" on
```java
    @ManyToOne(fetch = FetchType.LAZY, optional = false) 
private Test test;
```
exception goes from 
com.fasterxml.jackson.datatype.hibernate4.HibernateProxySerializer#findProxied
```java return init.getImplementation(); ```

Is it expected?

**Comment by cowtowncoder** [comment]
The first question usually is whether you can reproduce the behavior without spring, just using Jackson and Hibernate module. My guess is that the problem is that Spring controller is trying to serialize things after Hibernate session is closed; if so, this is not something Jackson has any control over.

**Comment by bademux** [comment]
exactly, but Hibernate4 Jackson module trying to handle this (at least for Collections - PersistentCollectionSerializer).
Let me ask it in other way:
Why only collection handled in such way? (I see here inconsistency with single reference, o at least undocumented behavioud)
Why hibernate session is opened per object property not per object itself?

**Comment by cowtowncoder** [comment]
Jackson module does not open sessions as far as I recall. It assumes a session exists, during processing.

Having said that, I do not use Hibernate nor Spring, so someone else may have to help here.
I only help to maintain this module.

**Comment by bademux** [comment]
Session id opened by hibernate added here: e4b3654067cc60d0c300ec2979896b87725175ff , see for more info:
```java
PersistentCollectionSerializer#openTemporarySessionForLoading()
```


Hibernate module opens session for read for *each* property collection, but ignoring simple residencies.

Please, consider to mark it as a bug if you find inconsistency in the behavior.

cast @eoger :)

**Comment by cowtowncoder** [comment]
I stand corrected. Thank you for pointing this out @bademux.
